### PR TITLE
optimize kube-ovn-monitor and add db restore process

### LINKED ISSUE #1299: Automatically fix db in "inconsistent data" status
### Feature request

We meet errors that ovn-sb storage status is `inconsistent data` and ovn-northd stop translating flow from ovn-nb to ovn-sb, ovn-controller cannot claim portbingding to ovn-sb, then all flow updates are stopped.

It seems that some transactions violate the database constraints, however we are not aware about how it happens and can only manually rebuild the ovn-sb database. We need a new way to automatically fix the db status.

### PR DESCRIPTION
**Pull request by hongzhen-ma**
#### What type of this PR
Examples of user facing changes:
- API changes

####
optimize the implementation of kube-ovn-monitor

#### Which issue(s) this PR fixes:
Fixes #1299

### CODE REVIEW

File: dist/images/install.sh [hunk 1]
```diff
@@ -2935,6 +2937,52 @@ dbtool(){
         dbstatus)
           kubectl exec "$OVN_NB_POD" -n $KUBE_OVN_NS -c ovn-central -- ovn-appctl -t /var/run/ovn/ovnnb_db.ctl ovsdb-server/get-db-storage-status OVN_Northbound
           ;;
+        restore)
+          # set ovn-central replicas to 0
+          replicas=$(kubectl get deployment -n $KUBE_OVN_NS ovn-central -o jsonpath={.spec.replicas})
+          kubectl scale deployment -n $KUBE_OVN_NS ovn-central --replicas=0
```
Comment (zhangzujian) [inline dist/images/install.sh:2943]: Should we wait for the operation to be completed?

File: dist/images/install.sh [hunk 2]
```diff
@@ -2935,6 +2937,52 @@ dbtool(){
...
+          podNames=`kubectl get pod -n $KUBE_OVN_NS | grep ovs-ovn | awk '{print $1}'`
+          echo "first nodeIP is $firstIP"
+
+          i=0
+          for nodeIp in $nodeIps
+          do
+            for pod in $podNames
+            do
+              hostip=$(kubectl get pod -n $KUBE_OVN_NS $pod -o jsonpath={.status.hostIP})
+              if [ nodeIp==hostip ]; then
```
Comment (zhangzujian) [inline dist/images/install.sh:2960]: Missing `$` here.

File: dist/images/install.sh (hunk 1 above)
Comment (hongzhen-ma) [reply dist/images/install.sh:2943]: We need to move db file to /tmp in ovs-ovn pod. The ovs-ovn pod will be crashed if we wait enough time when I did test. That will result in the failure of the command.So don't wait here.

File: dist/images/install.sh (hunk 2 above)
Comment (hongzhen-ma) [reply dist/images/install.sh]: modified

**Review by zhangzujian — APPROVED** [review]

**Comment by zhangzujian** [comment]
Please change the title.

**Comment by hongzhen-ma** [comment]
已回合1.8和1.9分支

**Pull request merged**